model/change_point_detection.go

```diff
@@ -387,16 +398,46 @@ func GetTotalPagesForChangePointsGroupedByVersion(ctx context.Context, env cedar
 		if err != nil {
 			return 0, errors.Wrap(err, "Cannot decode response of getting count of change points grouped by version")
 		}
-		return int(math.Ceil(float64(res.Count) / float64(pageSize))), nil
+		return int(math.Ceil(float64(res.Count) / float64(args.PageSize))), nil
 	}
-	return 0, errors.New("Not able to get count of total changepoints matching query")
+	return 0, nil
 }
 
-func appendAfterBaseGetChangePointsByVersionAgg(projectId string, additionalSteps ...bson.M) []bson.M {
+func appendAfterBaseGetChangePointsByVersionAgg(args GetChangePointsGroupedByVersionOpts, additionalSteps ...bson.M) []bson.M {
+	matchStage := bson.M{
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoProjectKey):  args.ProjectId,
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoVariantKey):  bson.M{"$regex": args.VariantRegex},
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoVersionKey):  bson.M{"$regex": args.VersionRegex},
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTaskNameKey): bson.M{"$regex": args.TaskRegex},
+		bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoTestNameKey): bson.M{"$regex": args.TestRegex},
+	}
+
+	if args.ThreadLevels != nil {
+		matchStage[bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoArgumentsKey, "thread_level")] = bson.M{
+			"$in": args.ThreadLevels,
```

Comment:
see above about thread level.

Reply:
Ditto. We want to filter on them, but how we do it/the representation is up for review.

Reply:
Sorry for being unclear earlier. I'm not objecting to filtering it, just the wat that this implementation institutionalizes thread_level over other kinds of arguments/parameters on tests.